packages/esm-patient-search-app/src/patient-search-result/patient-search-result.component.tsx

```diff
@@ -1,56 +1,48 @@
-import React from 'react';
-import dayjs from 'dayjs';
+import React, { useEffect, useState, useCallback } from 'react';
 import styles from './patient-search-result.scss';
-import { interpolateString, ExtensionSlot, useConfig, navigate } from '@openmrs/esm-framework';
+import { ExtensionSlot, useConfig, fetchCurrentPatient, interpolateString, navigate } from '@openmrs/esm-framework';
 import { SearchedPatient } from '../types/index';
 
 const PatientSearchResults: React.FC<PatientSearchResultsProps> = ({ patients, hidePanel }) => {
+  const [fhirPatients, setFhirPatients] = useState([]);
   const config = useConfig();
 
-  function renderPatient(patient: SearchedPatient) {
-    const preferredIdentifier = patient.identifiers.find((i) => i.preferred) || patient.identifiers[0];
+  useEffect(() => {
+    const fhirPatientsBuffer = [];
+    patients.forEach((patient) => {
+      fetchCurrentPatient(patient.uuid).then((response) => {
+        fhirPatientsBuffer.push(response.data);
+        setFhirPatients([...fhirPatientsBuffer]);
+      });
+    });
+  }, [patients]);
 
+  const onClickSearchResult = useCallback((patientUuid) => {
+    navigate({
+      to: interpolateString(config.search.patientResultUrl, {
+        patientUuid: patientUuid,
+      }),
+    });
+    hidePanel();
+  }, []);
+
+  function renderPatient(patient) {
```

Comment:
Either make this a component or (better) just put it inside the `return` statement JSX.